src/main/java/org/spongepowered/api/entity/living/player/CooldownTracker.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.living.player;
+
+import org.spongepowered.api.item.ItemType;
+
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+
+/**
+ * Provides access to the item cooldowns of a {@link Player}.
+ */
+public interface CooldownTracker {
```

Comment:
Maybe perhaps even have the `Player` object available for retrieval? Since the implementation provides such a method (only case it doesn't is in single player).

Thoughts @Zidane ?

Reply:
I'm not sure I see a need as this is gotten from the 'Player' object.

Reply:
In certain circumstances where consumer code is passing a `CooldownTracker`, it would be useful being able to just use the simple getter, simply by passing a `CooldownTracker` instead of the pair of `Player` and `CooldownTracker` objects.

Reply:
If you need both why not just provide the player since you can get the CDT from the player?

Reply:
:-1:, I don't see a need for it.